Firebase/Database/Persistence/FLevelDBStorageEngine.m

```diff
@@ -672,7 +672,7 @@ - (NSData*) serializePrimitive:(id)value {
     return [data subdataWithRange:NSMakeRange(1, data.length - 2)];
 }
 
-- (id)fixDoubleParsing:(id)value {
+- (id)fixDoubleParsing:(id)value __attribute__((no_sanitize("float-cast-overflow"))) {
```

Comment:
A better fix here would be to cast through an `int64_t` rather than `long long`.

Reply:
Sorry I wasn't clear: I meant while removing the `__attribute__` you added. Casting through `int64_t` guarantees that the type is large enough to hold a `double` so the overflow becomes impossible.

Reply:
@wilhuff This change doesn't fix the sanitizer issue.  I believe that for any architecture we care about `long long` is always the same as `int64_t`. The undefined behavior is from casting the `int64_t` back to a `double`, not from the cast to the `int64_t`.

Reply:
![screen shot 2018-06-25 at 9 13 41 am](https://user-images.githubusercontent.com/73870/41862222-1ab8eed8-7858-11e8-9216-180d053c4682.png)